Lib/glyphsLib/builder/transformations/intermediate_layers.py

```diff
@@ -0,0 +1,118 @@
+import logging
+import uuid
+
+from fontTools.varLib.models import VariationModel, normalizeValue
+
+from glyphsLib.classes import GSLayer, GSNode, GSPath
+from glyphsLib.builder.axes import get_regular_master
+
+
+logger = logging.getLogger(__name__)
+
+
+def resolve_intermediate_components(font):
+    components_with_intermediate_layers = set()
+    all_intermediate_locations = set()
+    for glyph in font.glyphs:
+        for layer in glyph.layers:
+            if "coordinates" in layer.attributes:
+                # First, let's find glyphs with intermediate layers
+                # which have components which don't have intermediate layers
+                for shape in layer.components:
+                    ensure_component_has_sparse_layer(font, shape, layer)
+                # Later we will check if everyone who uses me as a component
+                # has the same intermediate layers as I do
+                components_with_intermediate_layers.add(glyph.name)
+                all_intermediate_locations.add(tuple(layer.attributes["coordinates"]))
+
+
+def simple_variation_model(font):
+    tags = [axis.axisTag for axis in font.axes]
+    locations = [x.axes for x in font.masters]
+    limits = {tag: (min(x), max(x)) for tag, x in zip(tags, (zip(*locations)))}
+    master_locations = []
+    default_location = get_regular_master(font).axes
+    for loc in locations:
+        this_loc = {}
+        for ix, axisTag in enumerate(tags):
+            axismin, axismax = limits[axisTag]
+            this_loc[axisTag] = normalizeValue(
+                loc[ix], (axismin, default_location[ix], axismax)
+            )
+        master_locations.append(this_loc)
+    return VariationModel(master_locations, axisOrder=tags), limits
+
+
+def ensure_component_has_sparse_layer(font, component, parent_layer):
+    tags = [axis.axisTag for axis in font.axes]
+    model, limits = simple_variation_model(font)
+    location = parent_layer.attributes["coordinates"]
+    normalized_location = {
+        axisTag: normalizeValue(
+            location[ix], (limits[axisTag][0], limits[axisTag][0], limits[axisTag][1])
```

Comment:
since you're now using the real default and not assume min==default, I think you should adapt this part as well so that 0.0 default actually means the same thing as in the VarModel